backend/app/services/aiod.py

```diff
@@ -0,0 +1,164 @@
+from enum import Enum
+from pathlib import Path
+from urllib.parse import urljoin
+
+import httpx
+from fastapi import HTTPException
+from pydantic import Json
+
+from app.config import settings
+from app.helpers import Pagination
+from app.schemas.dataset import Dataset
+from app.schemas.ml_model import MLModel
+
+
+class AssetType(Enum):
+    DATASETS: str = "datasets"
+    ML_MODELS: str = "ml_models"
+    PUBLICATIONS: str = "publications"
+    PLATFORMS: str = "platforms"
+
+
+class AIoDClientWrapper:
+    async_client = None
+
+    def start(self):
+        """Instantiate the client. Call from the FastAPI startup hook."""
+        self.async_client = httpx.AsyncClient(verify=settings.AIOD_API.VERIFY_SSL)
+
+    async def stop(self):
+        """Gracefully shutdown. Call from FastAPI shutdown hook."""
+        await self.async_client.aclose()
+        self.async_client = None
+
+    @property
+    def client(self):
+        assert self.async_client is not None
+        return self.async_client
+
+
+aiod_client_wrapper = AIoDClientWrapper()
```

Comment:
I understand the logic, but I'm still thinking about whether it would not be better to create this in main.py and then inject this through dependency injection (through Depends the routes would get an instance of aiod_client_wrapper that they would pass to these get_assets, get_asset, ... functions). Then, this global variable would be in main.py, where other globals live.

Reply:
I think we can discuss this offline. Nevertheless, e can still keep it for another PR as I did not change the logic of this part in this PR so far.